### DISCUSSION TITLE
文档描述不清, 希望能够照顾没用过CQ的开发

### DISCUSSION THREAD

**Discussion by L1yp** [howmanybots/onebot#140] [问答]
https://github.com/howmanybots/onebot/blob/master/v11/specs/message/segment.md#%E5%9B%BE%E7%89%87

[1] 发送时，file 参数除了支持使用收到的图片文件名直接发送外，还支持：

PCQQ的图片文件名是乱码  这个文件名怎么和文件路径/URL路径 做区分? 
通过scheme协议判断?

**Comment by kj415j45** [comment]
@L1yp 在CQ还在的时候，以文件名发送的本质是以CQ的图片文件夹为基的相对路径。即 `X:/path to CQ/data/image/`
语音为 `X:/path to CQ/data/record/`

**Comment by L1yp** [comment]
> @L1yp 在CQ还在的时候，以文件名发送的本质是以CQ的图片文件夹为基的相对路径。即 `X:/path to CQ/data/image/`
> 语音为 `X:/path to CQ/data/record/`

CQ会自动下载 图片吗?

**Comment by ExerciseBook** [comment]
当时 CQ 会缓存图片文件的一些信息。

**Comment by kanrichan** [comment]
go-cqhttp和cqhttp-mirai（需要设置）是会缓存图片信息的

**Comment by stdrc** [comment]
> PCQQ的图片文件名是乱码 这个文件名怎么和文件路径/URL路径 做区分?
> 通过scheme协议判断?

OneBot 的实现理论上是需要能知道你的 `file=乱码.jpg` 是它之前收到的图片的，这时候它会把收到的图片原样发出去；如果你需要发一个你自己的新图片，用绝对路径、URL、或 base64。
